sdk/nodejs/output.ts

```diff
@@ -479,7 +534,7 @@ export function secret<T>(val: Input<T | undefined>): Output<Unwrap<T | undefine
 function createSimpleOutput(val: any) {
     return new Output(
         new Set(),
-        Promise.resolve(val),
+        val instanceof Promise ? val : Promise.resolve(val),
```

Comment:
Really a shame that this is needed - would have hoped that `Promise.resolve` does this itself internally - but looks like it's explicitly spec'd not to.

Reply:
Note: i'm keeping this for this PR.  However, this should not be necessary.  `Promise.Resolve(someP)` will just return `someP` if it is a Promise.  I've tested this on node 8/10/12 and validated that this is the case.  I'll open a bug to track removing this (but making sure we don't regress perf).